app/Http/Controllers/DashboardController.php

```diff
@@ -151,27 +153,25 @@ public function index()
             $network = $user->networks;
         }
 
+        // Look for groups where user ID exists in pivot table.  We have to explicitly test on deleted_at because
+        // the normal filtering out of soft deletes won't happen for joins.
+        $your_groups = Group::join('users_groups', 'users_groups.group', '=', 'groups.idgroups')
+            ->leftJoin('events', 'events.group', '=', 'groups.idgroups')
+            ->where('users_groups.user', $user->id)
+            ->whereNull('users_groups.deleted_at')
+            ->orderBy('groups.name', 'ASC')
+            ->groupBy('groups.idgroups')
+            ->select([ 'groups.idgroups' , 'groups.name' ])
+            ->take(3)
+            ->get();
+
         return view('dashboard.index', [
-            'show_getting_started' => ! $userExistsInDiscourse || ! $has_profile_pic || ! $has_skills || ! $in_group || ! $in_event,
-            'gmaps' => true,
             'user' => $user,
-            'header' => true,
-            'user_exists_in_discourse' => $userExistsInDiscourse,
-            'in_group' => $in_group,
             'groupsNearYou' => $groupsNearYou,
-            'has_skills' => $has_skills,
-            'in_event' => $in_event,
-            'has_profile_pic' => $has_profile_pic,
-            'past_events' => $past_events,
             'upcoming_events' => $upcoming_events,
-            'outdated_groups' => $outdated_groups,
-            'inactive_groups' => $inactive_groups,
-            'news_feed' => $news_feed,
-            'all_groups' => $all_groups,
-            'onboarding' => $onboarding,
-            'impact_stats' => $impact_stats,
-            'wiki_pages' => $wiki_pages,
-            'hot_topics' => $this->getDiscourseHotTopics(),
+            'topics' => $this->getDiscourseHotTopics()['talk_hot_topics'],
```

Comment:
I believe the intention was for this to change to latest topics, as opposed to hot topics - though perhaps this comes in part 2 of these two part stories.

Reply:
I did wonder about that.  

The Workbench page shows the latest ones, via MicrotaskingController::getDiscussionTopics.  Where would be a good place to move that method to?

Reply:
If it's logic that's going to be shared between two controllers, I'd probably create a service class, Services/DiscourseService.php, that's injected in to each of the controllers.

That said, on Workbench it is getting the latest topics for a specific tag, not the latest topics for all of Discourse which I think is the intention here, so it might need a bit of rejigging to accommodate that (or a new method).